cyhy/db/database.py

```diff
@@ -1582,15 +1582,26 @@ def get_indices(self):
 
     def save(self, *args, **kwargs):
         # Calculate severity from cvss on save
+        # Source: https://nvd.nist.gov/vuln-metrics/cvss
         cvss = self["cvss_score"]
-        if cvss == 10:
-            self["severity"] = 4
-        elif cvss >= 7.0:
-            self["severity"] = 3
-        elif cvss >= 4.0:
-            self["severity"] = 2
-        else:
-            self["severity"] = 1
+        if self["cvss_version"] == "2.0":
+            if cvss == 10:
+                self["severity"] = 4
+            elif cvss >= 7.0:
+                self["severity"] = 3
+            elif cvss >= 4.0:
+                self["severity"] = 2
+            else:
+                self["severity"] = 1
+        elif self["cvss_version"] in ["3.0", "3.1"]:
+            if cvss >= 9.0:
+                self["severity"] = 4
+            elif cvss >= 7.0:
+                self["severity"] = 3
+            elif cvss >= 4.0:
+                self["severity"] = 2
+            else:
+                self["severity"] = 1
```

Comment:
So a couple of things to think about... 

# CVSS v3 Score `0`

CVSS v3 does have a defined severity of `None` when the CVSS score is `0`:
https://www.first.org/cvss/specification-document#Qualitative-Severity-Rating-Scale

Should we map this explicitly?

# Strict mapping 

The severity to score mapping is not continous: https://nvd.nist.gov/vuln-metrics/cvss

I've copied the table below:

CVSS v2.0 Severity | CVSS v2.0 Score | CVSS v3.0 Severity | CVSS v3.0 Score |
| -- | -- | -- | -- |
|  |   | None | 0.0 |
|Low | 0.0-3.9 | Low | 0.1-3.9|
|Medium | 4.0-6.9 | Medium | 4.0-6.9|
|High | 7.0-10.0 | High | 7.0-8.9|
|  |   | Critical | 9.0-10.0|

That is, there are gaps in their mapping function.  For example, `8.9` is high, `9.0` is critical, but `8.95` is undefined.  

There is a `roundup` function defined in the [CVSS equation documentation](https://www.first.org/cvss/specification-document#CVSS-v3-1-Equations):

> Roundup returns the smallest number, specified to 1 decimal place, that is equal to or higher than its input. For example, Roundup (4.02) returns 4.1; and Roundup (4.00) returns 4.0. To ensure consistent results across programming languages and hardware, see Appendix A for advice to Implementers on avoiding small inaccuracies introduced in some floating point implementations.

See also: https://www.first.org/cvss/specification-document#Appendix-A---Floating-Point-Rounding

So it seems that we should never see a `8.95` score, and if we ever did your logic would make the most sense.

Maybe a comment to that effect would be in order.

Reply:
re: CVSS severity mapping - I added a comment about this in https://github.com/cisagov/cyhy-core/pull/69/commits/ea1a9dc14ea590f7c4c55dbd87f8180a790d321e.  Let me know if that covers your concerns.

re: CVSS v3 Score `0` - For what it's worth, I queried our production dataset and confirmed that none of our vuln scans have a `cvss3_base_score` less than `3.1`.  Also, I'm not sure if I trust all of our CyHy and reporting code to function properly if it were to encounter a severity of 0 (or greater than 4 for that matter).  I took a whack at addressing this in a comment in https://github.com/cisagov/cyhy-core/pull/69/commits/ea1a9dc14ea590f7c4c55dbd87f8180a790d321e.  Let me know if that works for you.